packages/calcite-components/src/components/dropdown/dropdown.tsx

```diff
@@ -380,32 +372,65 @@ export class Dropdown
   calciteInternalDropdownItemKeyEvent(event: CustomEvent<ItemKeyboardEvent>): void {
     const { keyboardEvent } = event.detail;
     const target = keyboardEvent.target as HTMLCalciteDropdownItemElement;
+    let focusTarget;
 
     switch (keyboardEvent.key) {
       case "Tab":
         this.open = false;
         this.updateTabIndexOfItems(target);
         break;
       case "ArrowDown":
-        focusElementInGroup(this.items, target, "next");
+        focusTarget = focusElementInGroup(this.items, target, "next");
         break;
       case "ArrowUp":
-        focusElementInGroup(this.items, target, "previous");
+        focusTarget = focusElementInGroup(this.items, target, "previous");
         break;
       case "Home":
-        focusElementInGroup(this.items, target, "first");
+        focusTarget = focusElementInGroup(this.items, target, "first");
         break;
       case "End":
-        focusElementInGroup(this.items, target, "last");
+        focusTarget = focusElementInGroup(this.items, target, "last");
         break;
     }
 
+    if (focusTarget) {
+      this.triggers.forEach((trigger) => {
+        trigger.setAttribute("aria-activedescendant", focusTarget.id);
+      });
+    }
+
     event.stopPropagation();
   }
 
   @Listen("calciteInternalDropdownItemSelect")
   handleItemSelect(event: CustomEvent<RequestedItem>): void {
+    const { selectionMode } = event.detail.requestedDropdownItem;
+    const selectionArray = [
+      event.detail.requestedDropdownGroup.id,
+      event.detail.requestedDropdownItem.id,
+    ];
+    let newSelection = [];
+
     this.updateSelectedItems();
+
+    this.triggers.forEach((trigger) => {
```

Comment:
nitpick: maybe this logic can be moved into a function so its apparent what its goal is?

Something like `private _setSelectionAria()`? I think thats what its doing right?